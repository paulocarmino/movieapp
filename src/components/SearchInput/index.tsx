import React from 'react'
import { TbSearch } from 'react-icons/tb'

const SearchInput = React.forwardRef<HTMLInputElement>(({ ...rest }, ref) => {
  return (
    <div className="flex items-center w-full">
      <div className="w-full">
        <label htmlFor="search" className="sr-only">
          Search
        </label>
        <div className="relative">
          <div className="flex absolute inset-y-0 left-0 items-center pl-3 pointer-events-none">
            <TbSearch className="w-5 h-5 text-gray-400" aria-hidden="true" />
          </div>
          <input
            ref={ref}
            id="search"
            name="search"
            className="block py-2 pr-3 pl-10 w-full text-sm placeholder:text-gray-500 text-gray-900 focus:text-gray-900 bg-white rounded-md border border-gray-300 focus:outline-none sm:text-sm"
            placeholder="Search movies..."
            type="search"
            {...rest}
          />
        </div>
      </div>
    </div>
  )
})

export default SearchInput
