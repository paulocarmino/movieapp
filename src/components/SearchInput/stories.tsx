import { Story, Meta } from '@storybook/react'
import SearchInput from '.'

export default {
  title: 'SearchInput',
  component: SearchInput,
  parameters: {
    backgrounds: {
      default: 'dark'
    }
  }
} as Meta

export const Default: Story = () => <SearchInput />
