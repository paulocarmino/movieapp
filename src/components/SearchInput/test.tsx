import { render, screen } from '@testing-library/react'

import SearchInput from '.'

describe('<SearchInput />', () => {
  it('deve exibir o component corretamente', () => {
    const { container } = render(<SearchInput />)

    expect(screen.getByPlaceholderText('Search movies...')).toBeInTheDocument()

    expect(container.firstChild).toMatchSnapshot()
  })
})
