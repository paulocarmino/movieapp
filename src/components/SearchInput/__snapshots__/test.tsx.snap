// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SearchInput /> deve exibir o component corretamente 1`] = `
<div
  class="flex items-center"
>
  <div
    class="w-full"
  >
    <label
      class="sr-only"
      for="search"
    >
      Search
    </label>
    <div
      class="relative"
    >
      <div
        class="flex absolute inset-y-0 left-0 items-center pl-3 pointer-events-none"
      >
        <svg
          aria-hidden="true"
          class="w-5 h-5 text-gray-400"
          fill="none"
          height="1em"
          stroke="currentColor"
          stroke-linecap="round"
          stroke-linejoin="round"
          stroke-width="2"
          viewBox="0 0 24 24"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <desc />
          <path
            d="M0 0h24v24H0z"
            fill="none"
            stroke="none"
          />
          <circle
            cx="10"
            cy="10"
            r="7"
          />
          <line
            x1="21"
            x2="15"
            y1="21"
            y2="15"
          />
        </svg>
      </div>
      <input
        class="block py-2 pr-3 pl-10 w-full text-sm placeholder:text-gray-500 focus:text-gray-900 focus:placeholder:text-gray-400 bg-white rounded-md border border-gray-300 focus:outline-none sm:text-sm"
        id="search"
        name="search"
        placeholder="Search movies..."
        type="search"
      />
    </div>
  </div>
</div>
`;
